modules/src/main/assembly/dist.xml

```diff
@@ -13,119 +13,33 @@
-             <filtered>true</filtered>
-        </file>
-        <file>
-             <source>${module.xml.basedir}/ogm/cassandra/module.xml</source>
-             <outputDirectory>/org/hibernate/ogm/cassandra/${hibernate.ogm.cassandra.module.slot}</outputDirectory>
-             <filtered>true</filtered>
-        </file>
-        <file>
-             <source>${module.xml.aliases.basedir}/ogm/cassandra/module.xml</source>
-             <outputDirectory>/org/hibernate/ogm/cassandra/main</outputDirectory>
-             <filtered>true</filtered>
-        </file>
-        <file>
-             <source>${module.xml.basedir}/ogm/cassandra-driver/module.xml</source>
-             <outputDirectory>/org/hibernate/ogm/cassandra/driver/${hibernate.ogm.cassandra.module.slot}</outputDirectory>
-             <filtered>true</filtered>
-        </file>
-        <file>
-             <source>${module.xml.basedir}/ogm/redis/module.xml</source>
-             <outputDirectory>/org/hibernate/ogm/redis/${hibernate.ogm.redis.module.slot}</outputDirectory>
-             <filtered>true</filtered>
-        </file>
-        <file>
-             <source>${module.xml.aliases.basedir}/ogm/redis/module.xml</source>
-             <outputDirectory>/org/hibernate/ogm/redis/main</outputDirectory>
-             <filtered>true</filtered>
-        </file>
-        <file>
-             <source>${module.xml.basedir}/ogm/redis-driver/module.xml</source>
-             <outputDirectory>/org/hibernate/ogm/redis/driver/${hibernate.ogm.redis.module.slot}</outputDirectory>
+             <source>${module.xml.aliases.basedir}/ogm/ignite/module.xml</source>
+             <outputDirectory>/org/hibernate/ogm/ignite/main</outputDirectory>
              <filtered>true</filtered>
         </file>
     </files>
 
     <dependencySets>
-        <!-- OGM -->
         <dependencySet>
-             <useProjectArtifact>false</useProjectArtifact>
-             <outputDirectory>org/hibernate/ogm/ehcache/${hibernate.ogm.ehcache.module.slot}</outputDirectory>
-             <useTransitiveFiltering>false</useTransitiveFiltering>
-             <unpack>false</unpack>
-             <includes>
-                 <include>org.hibernate.ogm:hibernate-ogm-ehcache</include>
-                 <include>net.sf.ehcache:ehcache-core</include>
-              </includes>
+            <useProjectArtifact>false</useProjectArtifact>
+            <outputDirectory>org/hibernate/ogm/ignite/${hibernate.ogm.ignite.module.slot}</outputDirectory>
+            <useTransitiveFiltering>false</useTransitiveFiltering>
+            <unpack>false</unpack>            
+            <includes>
+                <include>org.hibernate.ogm:hibernate-ogm-ignite</include>
+                <include>org.apache.ignite:*</include>
+                <include>javax.cache:cache-api</include>
+                <include>org.gridgain:ignite-shmem</include>
+                <include>commons-codec:commons-codec</include>
```

Comment:
I think this 3 libraries should be in a module ignite-internal, this way they won't interfere with the app using the module. We do something similar for neo4j, check the module neo4j-internal for inspiration.

About commons-codec, maybe you can use the version already on WildFly? This way we can avoid adding the jar in the module.

What's ignite-shmem?

Reply:
About commons-codec. Wildfly contains 1.10 version. Apache Ignite required 1.16. It can be critical. But ... But I will try

Reply:
Don't worry about it, let's keep the ignite version. As long as it is in an internal module it won't affect the application using hibernate-ogm-ignite

Reply:
About ignite-shmem. See https://mvnrepository.com/artifact/org.gridgain/ignite-shmem .